packages/select/src/util/StylesBuilder.ts

```diff
@@ -71,7 +71,7 @@ const resolveInputBorderColor = (
   variant: SelectVariant | undefined
 ): string | undefined => {
   if (isDisabled) {
-    return colors.borderColor;
+    return colors.disabledBackgroundColor;
```

Comment:
This functions returns border color, it cannot return background color, even if those colors happen to be the same.
The different variables have roles, use them based on their roles, not their values.

Reply:
The correct solution here is to add `colors.disabledBorderColor` (which doesn't exist today), and set it to the same color as the background.